extensions/sdk/src/Rally.ts

```diff
@@ -137,13 +143,25 @@ export class Rally {
    * Attempt to fetch the attribution codes from the store page URL for this extension.
    */
   private async storeAttributionCodes() {
+    const url = this._options.functionsHost;
+    url.pathname = "offboard";
```

Comment:
We don't have an offboard function yet on the new functions host, right? The old one is connected to a separate Google Cloud instance, I thought?

We should also see about having this offboard function unenroll the user from the study in Firebase, I think.

Reply:
That's right, we need to also stand up this function.

I don't think we should unenroll the user though, that's something they can log into their account to do. I think the extension being uninstalled on one device shouldn't unenroll them everywhere (it could be installed on another device or browser for instance).

Reply:
ok cool and yes that makes sense re: unenrollment. I think we should make the function part of the same PR - maybe make this a draft PR for now?

Also, I wonder what are the chances that the attribution codes are still lingering around in local storage if the user is uninstalling after months. It seems that ideally we'd record an uninstallation ping, and have the original attribution code picked up from the environment?

Reply:
> ok cool and yes that makes sense re: unenrollment. I think we should make the function part of the same PR - maybe make this a draft PR for now?

Good idea, I made it a draft for now. I feel like at minimum we want to grab the attribution codes and redirect the user somewhere, maybe similar to https://rally.mozilla.org/leaving-rally/index.html but the text on that is wrong.

> Also, I wonder what are the chances that the attribution codes are still lingering around in local storage if the user is uninstalling after months. It seems that ideally we'd record an uninstallation ping, and have the original attribution code picked up from the environment?

It should be stored permanently in local storage, but I think the real problem is (if we go that route) how do we identify a particular user? When we did this for the core add-on, we didn't want to pass the user's rally ID on a URL like this (just out of an abundance of caution since it's PII). What we did was generate a deletion ID and send that in with the enrollment ping, and it was only used for this purpose.

Since we don't want to generate a deletion ping though, I think all we really care about is the UTM codes. I like this approach because it avoids using explicit PII. Instead, it just lets us know which attribution codes are correlated with uninstalls, but it doesn't identify the specific user that uninstalled.

Reply:
Yeah I was thinking maybe if it's an authenticated HTTP function we'd maybe have access to the Firebase user doc? but if we don't need to identify the user then yeah I agree UTM + redirect will be fine.

Reply:
Hm @aaga yeah it's hard to say, this uninstall URL will fire no matter if the user is logged in or not. We could change the URL depending on the login status I guess, but I think that's probably more complicated than is warranted. It's probably safer to assume they aren't logged in.